docs/src/content/docs/reference/plugins.md

```diff
@@ -161,3 +162,50 @@ The example above will log a message that includes the provided info message:
 ```shell
 [long-process-plugin] Starting long process…
 ```
+
+#### `injectTranslations`
+
+**type:** `(translations: Record<string, Record<string, string>>) => void`
+
+A callback function to add or update translations strings.
+
+In the following example, the plugin injects a custom UI string translation named `myPlugin.doThing` for the `en` locale and `fr` locales:
+
+```ts {6-13} /(injectTranslations)[^(]/
+// plugin.ts
+export default {
+  name: 'plugin-with-translations',
+  hooks: {
+    setup({ injectTranslations }) {
+      injectTranslations({
+        en: {
+          'myPlugin.doThing': 'Do the thing',
+        },
+        fr: {
+          'myPlugin.doThing': 'Faire le truc',
+        },
+      });
+    },
+  },
+};
+```
+
+To use the injected translation in your plugin UI using the `locals.t` object, please check the [“Accessing UI strings” guide](/guides/i18n/#accessing-ui-strings) for more information.
+
+To type the `locals.t` object in the context of your plugin, declare the following global namespaces in a TypeScript declaration file:
+
+```ts
+// env.d.ts
```

Comment:
By default, no, just like our `virtual.d.ts` file for example where we don't do anything special for it. You could definitely setup your TS config so that it does not, but in this case, I assume it would be up to the author to handle that? I'm also not confident we could even document all cases.

Reply:
OK, perfect. If nothing else in theory is needed, no need to add anything. That’s great.